### ISSUE TITLE
Output of system test looks suspicious

### ISSUE THREAD

**Issue by rhuss** [syndesisio/syndesis#676]
Although the systems succeeds, the output looks suspicious to me:

* Why does the test create a new project to run it ? I thought it was reusing on of the pool projects ?
* Why is syndesis-ouathproxy still in status 'ContainerCreating' but the test succeeds ?
* Why is the template tried to be applied a second time at the end of this test (which fails because the temp project is deleted) ?
* How do we deal with the OAuth client handling ? (see error when applying the template the first time)

Any ideas @iocanel ?

```
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Initializing Session:c3bd8c5a
Using Kubernetes at: https://api.rh-idev.openshift.com:443/
Creating project: itest-c3bd8c5a
To switch to the new project: oc project itest-c3bd8c5a
Executing environment setup script from:file:/workspace/app/tests/target/test-classes/setup.sh
Installing Syndesis in itest-c3bd8c5a from: https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-ci.yml
Now using project "itest-c3bd8c5a" on server "https://api.rh-idev.openshift.com:443".
Local ImageStreams will be used
template "syndesis-ci" created
--> Deploying template "itest-c3bd8c5a/syndesis-ci" to project itest-c3bd8c5a

     syndesis-ci
     ---------
     Syndesis is deployed to itest-c3bd8c5a.b6ff.rh-idev.openshiftapps.com.

     * With parameters:
        * ROUTE_HOSTNAME=itest-c3bd8c5a.b6ff.rh-idev.openshiftapps.com
        * OPENSHIFT_MASTER=https://api.rh-idev.openshift.com:443
        * OPENSHIFT_OAUTH_CLIENT_ID=itest-c3bd8c5a
        * OPENSHIFT_OAUTH_CLIENT_SECRET=G1WYMNEKvBhY3fNDvBjDmusT712P6SLQccqjmX1M6oOtadomijxfA1LQ3HQfe8G2 # generated
        * OPENSHIFT_OAUTH_DEFAULT_SCOPES=user:full
        * Memory Limit=255Mi
        * Namespace=openshift
        * PostgreSQL Connection Username=syndesis
        * PostgreSQL Connection Password=yFXytN3DfPjgqnhS # generated
        * PostgreSQL Database Name=syndesis
        * Volume Capacity=1Gi
        * PostgreSQL SampleDB Connection Password=o46RhvCl5pqG4YY7 # generated
        * Test Support Enabled=true
        * Demo Data Enabled=false
        * Syndesis Image Registry=docker.io
        * Enable Integration Deployment=true
        * Image Stream Namespace=
        * OAuth Cookie Secret=P8vH0pqTMANhoVAkqqK5bfUnLw32TCs4 # generated
        * SYNDESIS_ENCRYPT_KEY=tT6q73h3cQCeskdRNX3cSyIfO6XrfTlgVox2u6bWRJbeQBiA1BdkiCbHR8Qy4YOE # generated

--> Creating resources ...
    imagestream "syndesis-rest" created
    imagestream "syndesis-ui" created
    imagestream "atlasmap" created
    imagestream "syndesis-verifier" created
    imagestream "oauth-proxy" created
    imagestream "s2i-java" created
    secret "syndesis-oauth-proxy-cookie-secret" created
    service "syndesis-ui" created
    deploymentconfig "syndesis-ui" created
    configmap "syndesis-ui-config" created
    serviceaccount "syndesis-atlasmap" created
    service "syndesis-atlasmap" created
    deploymentconfig "syndesis-atlasmap" created
    configmap "syndesis-atlasmap-config" created
    configmap "syndesis-sampledb-config" created
    service "syndesis-db" created
    persistentvolumeclaim "syndesis-db" created
    deploymentconfig "syndesis-db" created
    service "syndesis-oauthproxy" created
    route "syndesis" created
    deploymentconfig "syndesis-oauthproxy" created
    serviceaccount "syndesis-rest" created
    serviceaccount "syndesis-integration" created
    service "syndesis-rest" created
    route "syndesis-rest" created
    deploymentconfig "syndesis-rest" created
    configmap "syndesis-rest-config" created
    service "syndesis-verifier" created
    deploymentconfig "syndesis-verifier" created
    configmap "syndesis-verifier-config" created
    service "todo" created
    route "todo" created
    imagestream "todo" created
    buildconfig "todo" created
    deploymentconfig "todo" created
    error: User "system:serviceaccount:syndesis-ci:jenkins" cannot create oauthclients.oauth.openshift.io at the cluster scope
    rolebinding "syndesis:editors" created
--> Failed
Did not find any kubernetes configuration.
Replication controller: [syndesis-atlasmap-1]
Replication controller: [syndesis-db-1]
Replication controller: [syndesis-oauthproxy-1]
Replication controller: [syndesis-rest-1]
Replication controller: [syndesis-ui-1]
Replication controller: [syndesis-verifier-1]
Replication controller: [todo-1]
Pod: [syndesis-atlasmap-1-tm1jk] Status: [Running]
Pod: [syndesis-db-1-6ggcr] Status: [Running]
Pod: [syndesis-oauthproxy-1-deploy] Status: [Running]
Pod: [syndesis-oauthproxy-1-h3474] Status: [Pending]
Pod: [syndesis-rest-1-r8c6p] Status: [Running]
Pod: [syndesis-ui-1-6qdqj] Status: [Running]
Pod: [syndesis-ui-1-deploy] Status: [Running]
Pod: [syndesis-verifier-1-2wjdj] Status: [Running]
Pod: [todo-1-build] Status: [Succeeded]
Pod: [todo-1-qh9vh] Status: [Running]
Service: [syndesis-atlasmap] IP: [172.30.244.46] Ports: [ 80 ]
Service: [syndesis-db] IP: [172.30.33.213] Ports: [ 5432 ]
Service: [syndesis-oauthproxy] IP: [172.30.169.77] Ports: [ 8443 ]
Service: [syndesis-rest] IP: [172.30.1.79] Ports: [ 80 ]
Service: [syndesis-ui] IP: [172.30.125.163] Ports: [ 80 ]
Service: [syndesis-verifier] IP: [172.30.235.31] Ports: [ 80 ]
Service: [todo] IP: [172.30.72.236] Ports: [ 8080 ]
[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 87.827 s - in io.syndesis.systests.DeploymentTest
[INFO] Running io.syndesis.systests.SyndesisRestTest
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 s - in io.syndesis.systests.SyndesisRestTest
Deleting project: itest-c3bd8c5a...
Project: itest-c3bd8c5a, successfully deleted
Executing environment teardown script from:file:/workspace/app/tests/target/test-classes/teardown.sh
Remove Syndesis from itest-c3bd8c5a
Already on project "itest-c3bd8c5a" on server "https://api.rh-idev.openshift.com:443".
Displaying the list of Pods in project: itest-c3bd8c5a
NAME                           READY     STATUS              RESTARTS   AGE
syndesis-atlasmap-1-tm1jk      1/1       Running             0          45s
syndesis-db-1-6ggcr            1/1       Running             0          1m
syndesis-oauthproxy-1-deploy   1/1       Running             0          1m
syndesis-oauthproxy-1-h3474    0/1       ContainerCreating   0          25s
syndesis-rest-1-r8c6p          1/1       Running             0          1m
syndesis-ui-1-6qdqj            1/1       Running             0          1m
syndesis-ui-1-deploy           1/1       Running             0          1m
syndesis-verifier-1-2wjdj      1/1       Running             0          1m
todo-1-build                   0/1       Completed           0          1m
todo-1-qh9vh                   1/1       Running             0          21s
Removing Syndesis from itest-c3bd8c5a
error: error processing the template "syndesis-ci": processedtemplates "syndesis-ci" is forbidden: unable to create new content in namespace itest-c3bd8c5a because it is being terminated.
No resources found
Destroying Session:c3bd8c5a
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 3, Failures: 0, Errors: 0, Skipped: 2
```

**Comment by rhuss** [comment]
The link to this testouput --> https://1872-105563335-gh.circle-artifacts.com/0/workspace/test_log.txt

**Comment by iocanel** [comment]
**Why does the test create a new project to run it ? I thought it was reusing on of the pool projects ?**

Arquillian will create a new project, unless we explicitly tell it to use an existing one. We used to do so by specifying an environment variable named `NAMESPACE_USE_EXISTING` which was removed as of: https://github.com/syndesisio/syndesis/commit/107e4cd8c38baaed6436fd36710e6ccaf63d453d#diff-b8af0ec63d5d4bb827caa04c8f3b6c04L570

**Why is syndesis-ouathproxy still in status 'ContainerCreating' but the test succeeds ?**

The system tests, don't do much at the moment. They deploy the project, they wait for some endpoints (explicitly specified) to become ready and they just send an http request to the ui. The proxy isn't involved in any of these.

**Why is the template tried to be applied a second time at the end of this test (which fails because the temp project is deleted) ?**

It is not applied for a second time. Its processed into a list and the output is passed to delete (for cleanup purposes). My understanding is that it fails due to a race.

Ideally, we shouldn't even use `oc` for those but we should just let arquillian-cube create then environment and cleanup for us. This would also allows us to `implicitly wait` (see previous comment).
This was blocked by: https://github.com/openshift/origin/issues/13197 


**How do we deal with the OAuth client handling ? (see error when applying the template the first time)**
The error is about trying to access cluster wide resources (which we don't have permissions to). We usually get away with it, because usually those are already there and we don't need to do much with them.
